llvm/lib/SYCLLowerIR/GlobalOffset.cpp

```diff
@@ -59,11 +59,21 @@ ModulePass *llvm::createGlobalOffsetPassLegacy() {
   return new GlobalOffsetLegacy();
 }
 
+// Recursive helper function to collect Loads from GEPs in a BFS fashion.
+static void getLoads(Instruction *P, SmallVectorImpl<Instruction *> &Traversed,
+                     SmallVectorImpl<LoadInst *> &Loads) {
+  Traversed.push_back(P);
```

Comment:
May be an assert at the very top of this function body to check for *p being either a load or a GEP might be better?

Reply:
I mean the check is already taking place in the `dyn_cast<LoadInst>(P) and the assertion if the else branch is taken: `assert(isa<GetElementPtrInst>(*P));`.
If the instruction is not a Load or GEP, the code will also throw without another assert in the first line, so I think it would add a sort of redundancy here.